src/lib.rs

```diff
@@ -213,6 +289,14 @@ impl<'a, T> ReadLock<'a, T> {
         &self.guard.data[ptr.index]
     }
 
+    /// Borrow a weak pointed component for reading.
+    pub fn access_weak(&self, pointer: &WeakPointer<T>) -> Result<&T, UpgradeErr> {
```

Comment:
I don't think `access_weak` brings anything on the table, really. It may seem useful, but you can do the same from user space and that would be as efficient, so I'd be in favour of reducing the API surface here